yellow-paper/docs/circuits/private-kernel-initial.md

```diff
@@ -0,0 +1,258 @@
+# Private Kernel Circuit - Initial
+
+:::info Disclaimer
+This is a draft. These requirements need to be considered by the wider team, and might change significantly before a mainnet release.
+:::
+
+## Requirements
+
+In the **initial** kernel iteration, the process involves taking a transaction request and private call data, verifying their integrity, and preparing the necessary data for subsequent circuits to operate. This step is particularly beneficial due to its separation from the [inner private kernel circuit](./private-kernel-inner.md), as the first call lacks a "previous kernel" to process. Additionally, it executes tasks that are pertinent to a transaction and need only occur once.
+
+### Key Responsibilities Specific to this Circuit
+
+#### Validating the correspondence of function call with caller's intent.
+
+This entails ensuring that the following data from the private call aligns with the specifications in the transaction request:
+
+- Contract address.
+- [Function data](#function_data).
+- Function arguments.
+
+> Although it's not enforced in the protocol, it is customary to provide a signature signed over the transaction request and verify it in the first function call. This practice guarantees that only the party possessing the key(s) can authorize a transaction with the exact transaction request.
+
+#### Verifying the legitimacy of the function as the entrypoint.
+
+- It must be a private function.
+- It must not be an internal function.
```

Comment:
Are internal functions defined anywhere? If not, we need to create an issue to define them. if they are, having `internal function` link to the relevant page would be v. helpful